src/lib/APS2.cpp

```diff
@@ -487,58 +528,98 @@ void APS2::write_memory(const uint32_t & addr, const vector<uint32_t> & data){
 	 * data = vector<uint32_t> data
 	 */
 
-	if (data.size() == 0) {
-		return;
-	}
```

Comment:
We're actually okay.  We're saved by the datagram chunking returning an empty vector and then there is nothing to do. Added test coverage for both in 0375f64c44d9babb980f6ad55f7462e46bcbab9c